docs/guide/controller-programming.md

```diff
@@ -51,29 +109,106 @@ When the loop exists, no further communication with Webots is possible and the o
 Now that we have seen how to print a message to the console, we shall see how to read the sensors of a robot.
 The next example does continuously update and print the value returned by a [DistanceSensor](../reference/distancesensor.md):
 
+%tab-component "language"
+%tab "C"
 ```c
 #include <webots/robot.h>
 #include <webots/distance_sensor.h>
 #include <stdio.h>
 
-#define TIME_STEP 32
-
 int main() {
   wb_robot_init();
 
-  WbDeviceTag ds = wb_robot_get_device("my_distance_sensor");
-  wb_distance_sensor_enable(ds, TIME_STEP);
+  WbDeviceTag sensor = wb_robot_get_device("my_distance_sensor");
+  wb_distance_sensor_enable(sensor, 32);
 
-  while (wb_robot_step(TIME_STEP) != -1) {
-    double dist = wb_distance_sensor_get_value(ds);
-    printf("sensor value is %f\n", dist);
+  while (wb_robot_step(32) != -1) {
+    const double value = wb_distance_sensor_get_value(sensor);
+    printf("Sensor value is %f\n", value);
   }
 
   wb_robot_cleanup();
 
```

Comment:
I started to remove those new lines since it is not consistent with the C++ examples, I guess if we want to free anything we should indicate it with a comment